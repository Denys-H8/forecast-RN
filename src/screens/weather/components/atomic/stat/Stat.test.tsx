import React from 'react';
import { render } from '@testing-library/react-native';
import Humidity from './Stat';

const categoryWind = 'wind';
const categorySunrise = 'sunrise';
const categorySunset = 'sunset';
const wrongCategory = 'wrong';
const text = 'Some text';

describe('Stat ', () => {
  test('should render Stat with properly category name', () => {
    const { toJSON, getByTestId } = render(<Humidity category={categoryWind} text={text} />);

    expect(toJSON()).toMatchSnapshot();
    expect(getByTestId('Stat.Icon')).not.toBeNull();
  });

  test('should render Stat with properly category name', () => {
    const { toJSON, getByTestId } = render(<Humidity category={categorySunrise} text={text} />);

    expect(toJSON()).toMatchSnapshot();
    expect(getByTestId('Stat.Icon')).not.toBeNull();
  });

  test('should render Stat with properly category name', () => {
    const { toJSON, getByTestId } = render(<Humidity category={categorySunset} text={text} />);

    expect(toJSON()).toMatchSnapshot();
    expect(getByTestId('Stat.Icon')).not.toBeNull();
  });

  test('should render Stat with not properly category name', () => {
    const { toJSON, getByTestId } = render(<Humidity category={wrongCategory} text={text} />);

    expect(toJSON()).toMatchSnapshot();
    expect(getByTestId('Stat.Icon')).not.toBeNull();
  });
});
