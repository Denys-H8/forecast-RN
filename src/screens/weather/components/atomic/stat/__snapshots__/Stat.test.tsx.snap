// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat  should render Stat with not properly category name 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <IconMock
    fill="#fff"
    height={32}
    testID="Stat.Icon"
    width={32}
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#fff",
          "fontSize": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    Some text
  </Text>
</View>
`;

exports[`Stat  should render Stat with properly category name 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <IconMock
    fill="#fff"
    height={26}
    testID="Stat.Icon"
    width={26}
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#fff",
          "fontSize": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    Some text
  </Text>
</View>
`;

exports[`Stat  should render Stat with properly category name 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <IconMock
    fill="#fff"
    height={28}
    testID="Stat.Icon"
    width={28}
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#fff",
          "fontSize": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    Some text
  </Text>
</View>
`;

exports[`Stat  should render Stat with properly category name 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <IconMock
    fill="#fff"
    height={28}
    testID="Stat.Icon"
    width={28}
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#fff",
          "fontSize": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    Some text
  </Text>
</View>
`;
